src/Commands/CIConfigureCommand.php

```diff
@@ -58,26 +55,32 @@ public function configureCI(
         // Get the build metadata from the Pantheon site. Fail if there is
         // no build metadata on the master branch of the Pantheon site.
         $buildMetadata = $this->retrieveBuildMetadata("{$site_name}.dev") + ['url' => ''];
-        $desired_url = "git@github.com:{$target_project}.git";
-        if (!empty($buildMetadata['url']) && ($desired_url != $buildMetadata['url'])) {
-            throw new TerminusException('The site {site} is already configured to test {url}; you cannot use this site to test {desired}.', ['site' => $site_name, 'url' => $buildMetadata['url'], 'desired' => $desired_url]);
+        
+        if (empty($buildMetadata['url'])) {
+            throw new TerminusException('The site {site} does not have the required build metadata. This command can only be used for sites that have been correctly initialized with build:project:create.', ['site' => $site_name]);
         }
 
-        // Get our authenticated credentials from environment variables.
-        $github_token = $this->getRequiredGithubToken();
-        $circle_token = $this->getRequiredCircleToken();
+        // Create a git repository service provider appropriate to the URL
+        $this->inferGitProviderFromUrl($buildMetadata['url']);
+        $target_project = $this->projectFromRemoteUrl($buildMetadata['url']);
 
-        $circle_env = $this->getCIEnvironment($site_name, $options);
-        // TODO: check to see if we need to downgrade to only use dev environment
+        // Initialize providers
+        $this->createProviders(null, $input->getOption('ci'));
```

Comment:
No need to add `$input` and `$output` to this command handler; `$options['ci']` will also hold this value.

Reply:
Done.